great_expectations/rule_based_profiler/attributed_resolved_metrics.py

```diff
@@ -56,49 +52,35 @@ def _detect_illegal_array_type_or_shape(values: MetricValues) -> bool:
         ):
             return True
 
+        value: MetricValue
         if isinstance(values, (list, tuple)):
             if len(values) > 0:
-                value: MetricValue = values[0]
-                # Python "None" is illegal as candidate for conversion into "numpy.ndarray" type.
-                if value is None:
-                    return True
-
-                # Pandas "DataFrame" and "Series" are illegal as candidates for conversion into "numpy.ndarray" type.
-                if isinstance(
-                    value,
-                    deep_filter_properties_iterable(
-                        properties=(
-                            pd.DataFrame,
-                            pd.Series,
-                            sqlalchemy_engine_Row,
-                            pyspark_sql_Row,
-                            set,
-                        )
-                    ),
-                ):
-                    return True
+                values = deep_filter_properties_iterable(properties=values)
 
+                values_iterator: Iterator
                 # Components of different lengths cannot be packaged into "numpy.ndarray" type (due to undefined shape).
                 if all(isinstance(value, (list, tuple)) for value in values):
-                    values_iterator: Iterator = iter(values)
-                    first_value_length: int = len(next(values_iterator))
-                    current_value: Sized[Any]
+                    if values is not None and len(values) > 0:
```

Comment:
You need a `None` check above line 62 since `for value in values` will crash if `values` is `None` which is one of the return types of `deep_filter_properties_iterable`